lib/index.js

```diff
@@ -110,10 +108,7 @@ async function writeFileAsync (filename, data, options = {}) {
     if (options.tmpfileCreated) {
       await options.tmpfileCreated(tmpfile)
     }
-    if (isTypedArray(data)) {
-      data = typedArrayToBuffer(data)
-    }
-    if (Buffer.isBuffer(data)) {
+    if (Buffer.isBuffer(data) || ArrayBuffer.isView(data)) {
```

Comment:
typed arrays and dataviews both return true on `isView`, fs can handle either.

Reply:
yeay, this was the kind of pr i wanted to see, but just a nit pick, is it necessary to have the `Buffer.isBuffer`?
isn't using only `ArrayBuffer.isView` sufficient enough?

Reply:
```js
> ArrayBuffer.isView(Buffer.from('42'))
true
```
sure enough.  Feel free to submit a pr to clean it up, I'm done for the day and won't be cutting the release for this repo till tomorrow.